src/backend/metal/src/command.rs

```diff
@@ -1211,21 +1227,82 @@ impl CommandBuffer {
 
     fn set_depth_stencil_desc(
         &mut self,
-        depth_stencil_desc: &metal::DepthStencilState,
+        depth_stencil_desc: &pso::DepthStencilDesc,
+        depth_stencil_raw: &metal::DepthStencilState,
     ) -> soft::RenderCommand {
-        self.state.depth_stencil_desc = Some(depth_stencil_desc.clone());
-        soft::RenderCommand::SetDepthStencilDesc(depth_stencil_desc.clone())
+        self.state.pipeline_depth_stencil = Some((depth_stencil_desc.clone(), depth_stencil_raw.clone()));
+        soft::RenderCommand::SetDepthStencilDesc(depth_stencil_raw.clone())
```

Comment:
I wonder how expensive those clones (of metal stuff) are

Reply:
Yeah, I don't know the answer to this either. We could possible optimize some of these, or perhaps pool these somewhere to reuse them...